File: pset6/sentimental/mario/more/mario.py
# Prints mario's half pyrramids

from cs50 import get_int


def main():
    # Takes the user input
    while True:
        n = get_int("Height : ")
        if (n <= 23 and n >= 0):
            break
    # Prints the half pyramids
    for i in range(1, n + 1):
        cupto = n - i  # Stores the number of spaces in that row
        for j in range(1, cupto + 1):
            print(" ", end="")
        for k in range(cupto + 1, n + 1):  # Prints the left half
            print("#", end="")
        print("  ", end="")
        for l in range(1, i + 1):  # Prints the right half
            print("#", end="")
        print()


if __name__ == "__main__":
    main()

File: pset1/mario/more/mario.c
// Prints mario's half pyramids

#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Takes the user input
    int n;
    do
    {
        n = get_int("Height : ");
    }
    while (n > 23 || n < 0); // Reprompts the user if correct value not given

    // Prints the half pyramids
    for (int i = 1; i <= n; i++)
    {
        int cupto = n - i; // Stores the number of spaces in that row
        for (int j = 1; j <= cupto; j++)
        {
            printf(" ");
        }
        for (int k = cupto + 1; k <= n; k++) // Prints the left half
        {
            printf("#");
        }
        printf("  ");
        for (int l = 1; l <= i; l++) // Prints the right half
        {
            printf("#");
        }
        printf("\n");
    }
}